### ISSUE TITLE
New function suggestions

### ISSUE THREAD

**Issue by dktapps** [zhuowei/MCPELauncher#779]
Hi,
I'd like to request the addition of the following functions:

```
Entity.isImmobile(entity)

Block.isFlammable(id)
Block.setFlammable(id)
Block.getExplosionResistance(id)
Block.getLightLevel(id,data)
Block.getLightOpacity(id,data)

Item.setStackSize(id,data,stackSize)
Item.getStackSize(id,data)
Item.getCategory(id,data)
Item.overrideDefaultName(id,data,newName)
Item.getStackedByData(id)
```

Thanks!

**Comment by zedman2000** [comment]
You can already do almost all of these with existing functions.

**Comment by dktapps** [comment]
I'm after the missing getters for the existing setters.

**Comment by zedman2000** [comment]
I do like the get / set stack size though...that would help although you can get a rough idea by looking through the list of entities and checking their coordinates.  Anything less than 0.5block away can be added together as a stack...again, close, but not what you are asking.

**Comment by zedman2000** [comment]
yeah, I'm not saying what you are asking for is bad, just that you can work around most of them for now - so don't let it hold your development up!  :-)

**Comment by dktapps** [comment]
@zedman2000 actually I meant that as in, get the max amount of that item you can have in 1 inventory slot, like 16 for signs, and the ability to change this

**Comment by dktapps** [comment]
@zedman2000 yeah I gotcha

**Comment by zedman2000** [comment]
@dktapps understood.  Maybe add a 1 sentence under the listed function names so we understand what you are after.  You can see how easily they can be misinterpreted.  Generally speaking, I will just code the limit into my mod using the wiki.  Regardless, if you just set a stack of signs to be any number up to 255, it will be set...you just can't use the game mechanic to do it as it will respect the "16" for signs.  Maybe that will help you work around for now?

**Comment by dktapps** [comment]
@zedman2000 yeah I guess. It'd be a little awkward but it would work :)
